File: typescript/tiktok/search-videos.ts
// How do I search TikTok videos by keyword?
// https://openhandle.dev/questions/how-do-i-search-tiktok-videos-by-keyword
// Run: node tiktok/search-videos.ts (Node 24 or newer)

import { OpenHandle } from '@openhandle/sdk';

const openhandle = new OpenHandle({ apiKey: process.env.OPENHANDLE_TEST_KEY! });

let page = await openhandle.tiktok.search.posts.list({ q: 'synthetic' });

while (true) {
    for (const video of page.data) console.log(video.author?.handle, video.metrics.views, video.caption);
    const next = await page.next();
    if (!next) break;
    page = next;
}

File: python/tiktok/search_videos.py
# How do I search TikTok videos by keyword?
# https://openhandle.dev/questions/how-do-i-search-tiktok-videos-by-keyword
# Run: python tiktok/search_videos.py

import os

from openhandle import OpenHandle

openhandle = OpenHandle(api_key=os.environ["OPENHANDLE_TEST_KEY"])

for video in openhandle.tiktok.search.posts.items(q="synthetic"):
    print(video["author"]["handle"], video["metrics"]["views"], video["text"])
